test/jdk/java/awt/Dialog/ChoiceModalDialogTest.java

```diff
@@ -0,0 +1,138 @@
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+public class ChoiceModalDialogTest {
+    static Frame f;
+    static Dialog d;
+    static volatile boolean keyOK;
+    static volatile boolean mouseOK;
+    static TextField tf;
+    static Choice c;
+
+    public static void main(String[] args) throws Exception {
+        Robot r;
+        try {
+            r = new Robot();
+            r.setAutoDelay(100);
+            EventQueue.invokeAndWait(() -> {
+                f = new Frame("Frame");
+                c = new Choice();
+                f.setBounds(100, 300, 300, 200);
+                f.setLayout(new FlowLayout());
+                tf = new TextField(3);
+                f.add(tf);
+
+                c.add("1");
+                c.add("2");
+                c.add("3");
+                c.add("4");
+                f.add(c);
+
+                tf.addFocusListener(new FocusAdapter() {
+                    public void focusLost(FocusEvent ev) {
+                        d = new Dialog(f, "Dialog", true);
+                        d.setBounds(300, 300, 200, 150);
+                        d.addKeyListener(new KeyAdapter() {
+                            public void keyPressed(KeyEvent ev) {
+                                keyOK = true;
+                            }
+                        });
+                        d.addMouseListener(new MouseAdapter() {
+                            public void mousePressed(MouseEvent ev) {
+                                mouseOK = true;
+                            }
+                        });
+                        d.setVisible(true);
+                    }
+                });
+
+                f.setVisible(true);
+                f.toFront();
+            });
+            r.delay(1000);
+            r.waitForIdle();
+            EventQueue.invokeAndWait(() -> {
+                r.mouseMove(tf.getLocationOnScreen().x + tf.getSize().width / 2,
+                        tf.getLocationOnScreen().y + tf.getSize().height / 2);
+            });
+            r.delay(500);
+            r.waitForIdle();
+            EventQueue.invokeAndWait(() -> {
+                r.mouseMove(c.getLocationOnScreen().x + c.getSize().width - 4,
+                        c.getLocationOnScreen().y + c.getSize().height / 2);
+                r.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                r.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            });
+            r.delay(500);
+            r.waitForIdle();
+            EventQueue.invokeAndWait(() -> {
+                r.mouseMove(d.getLocationOnScreen().x + d.getSize().width / 2,
+                        d.getLocationOnScreen().y + d.getSize().height / 2);
+                r.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                r.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+                r.keyPress(KeyEvent.VK_A);
+                r.keyRelease(KeyEvent.VK_A);
+
+            });
```

Comment:
This test fails  with `keyOK = false` due to missing delay after KeyEvent.
For test stability I think it might be better to convert  keyOK and mouseOK into CountDownLatch.

```
    private static volatile CountDownLatch keyOK = new CountDownLatch(1);
    private static volatile CountDownLatch mouseOK = new CountDownLatch(1);
```

```suggestion
                r.keyRelease(KeyEvent.VK_A);
            });
           r.waitForIdle(); 
           r.delay(500); 
```

Reply:
I don't think `CountDownLatch` is required here. Anyhow I have added some delays after keypress too. Still didn't get failure in CI system. Please check with updated test once and revert.

Reply:
Since you mentioned this test passes locally as well as on CI with the updated test changes, the open problemlist entry for macOS can be removed now, right?

`java/awt/Dialog/ChoiceModalDialogTest.java 8161475 macosx-all`

Reply:
Not sure if I can remove from problemList since the mentioned bug - https://bugs.openjdk.org/browse/JDK-8161475 is not yet closed. So retaining it since we are only open sourcing it now.